primitives/tests/vid/mod.rs

```diff
@@ -1,32 +1,35 @@
-use jf_primitives::vid::{VidError, VidResult, VidScheme};
-
 use ark_std::{
     println,
     rand::{seq::SliceRandom, CryptoRng, RngCore},
     vec,
 };
+use core::iter::zip;
+use jf_primitives::vid::{VidError, VidResult, VidScheme};
 
 /// Correctness test generic over anything that impls [`VidScheme`]
 ///
 /// `pub` visibility, but it's not part of this crate's public API
 /// because it's in an integration test.
 /// <https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests>
 pub fn round_trip<V, R>(
-    vid_factory: impl Fn(usize, usize) -> V,
+    vid_factory: impl Fn(usize, usize, usize) -> V,
     vid_sizes: &[(usize, usize)],
+    multiplicities: &[usize],
```

Comment:
This vid_factory has lost all pretence of abstraction now that it also takes a multiplicity arg. We could recover full abstraction here but it's probably not worth the trouble so don't worry about it.

Reply:
Yes, weaving in the multiplicity sort of ruined the nice abstraction.